### PR TITLE
Fix cAdivsor docker validation

### PR DESCRIPTION
**Pull request by timstclair**
- Synchronize `Validate` policy with docker `factory.go` checks
- Don't enforce `ExecutionDriver == "native"` for docker >= v1.11

### CODE REVIEW

**Comment by timstclair** [comment]
@k8s-bot test this

File: manager/manager.go [hunk 1]
```diff
@@ -233,7 +233,6 @@ func (self *manager) Start() error {
 		glog.Errorf("Registration of the raw container factory failed: %v", err)
 	}
 
-	// FIXME - delete?
```
Comment (vishh) [inline manager/manager.go]: why remove this?

**Comment by vishh** [comment]
A couple of comments, otherwise LGTM

**Comment by timstclair** [comment]
@k8s-bot test this

File: manager/manager.go (hunk 1 above)
Comment (timstclair) [inline manager/manager.go]: I didn't mean to include it in https://github.com/google/cadvisor/pull/1190 so I removed it here.

**Comment by timstclair** [comment]
@k8s-bot test this

**Comment by timstclair** [comment]
Finally got tests passing - the issue is that `ServerVersion` doesn't get set in the info for docker < 1.9. I fixed it by falling back to the version API, and filling it in from that.

@vishh PTAL

**Comment by timstclair** [comment]
Dropped https://github.com/google/cadvisor/pull/1192/commits/f9377267a833048649e62d5933aa24ac8627697a which had snuck in here.

**Pull request merged**